src/main/webapp/app/course/learning-paths/participate/learning-path-storage.service.ts

```diff
@@ -0,0 +1,211 @@
+import { Injectable } from '@angular/core';
+import { NgxLearningPathDTO, NodeType } from 'app/entities/competency/learning-path.model';
+
+/**
+ * This service is used to store the histories and recommendations of learning path participation for the currently logged-in user.
+ */
+@Injectable({ providedIn: 'root' })
+export class LearningPathStorageService {
+    private readonly learningPathHistories: Map<number, StorageEntry[]> = new Map();
+    private readonly learningPathRecommendations: Map<number, StorageEntry[]> = new Map();
+
+    /**
+     * Stores the lecture unit in the learning path's history.
+     *
+     * @param learningPathId the id of the learning path to which the new entry should be added
+     * @param lectureId the id of the lecture, the lecture unit belongs to
+     * @param lectureUnitId the id of the lecture unit
+     * @return the entry that is stored
+     */
+    storeLectureUnit(learningPathId: number, lectureId: number, lectureUnitId: number) {
+        const entry = new LectureUnitEntry(lectureId, lectureUnitId);
+        this.store(learningPathId, entry);
+        return entry;
+    }
+
+    /**
+     * Stores the exercise in the learning path's history.
+     *
+     * @param learningPathId the id of the learning path to which the new entry should be added
+     * @param exerciseId the id of the exercise
+     * @return the entry that is stored
+     */
+    storeExercise(learningPathId: number, exerciseId: number) {
+        const entry = new ExerciseEntry(exerciseId);
+        this.store(learningPathId, entry);
+        return entry;
+    }
+    private store(learningPathId: number, entry: StorageEntry) {
+        if (!entry) {
+            return;
+        }
```

Comment:
According to the type of your parameter, this should not be possible. Either leave it out or update the type accordingly